assets/js/modules/analytics-4/components/widgets/TopCitiesWidget.js

```diff
@@ -21,15 +21,97 @@
  */
 import PropTypes from 'prop-types';
 
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
+import {
+	DATE_RANGE_OFFSET,
+	MODULES_ANALYTICS_4,
+} from '../../datastore/constants';
+import { ZeroDataMessage } from '../../../analytics/components/common';
+import { numFmt } from '../../../../util';
+import { MetricTileTable } from '../../../../components/KeyMetrics';
+const { useSelect, useInViewSelect } = Data;
+
 export default function TopCitiesWidget( { Widget } ) {
+	const dates = useSelect( ( select ) =>
+		select( CORE_USER ).getDateRangeDates( {
+			offsetDays: DATE_RANGE_OFFSET,
+		} )
+	);
+
+	const topcCitiesReportOptions = {
+		...dates,
+		dimensions: [ 'city' ],
+		metrics: [ { name: 'totalUsers' } ],
+		orderby: [
+			{
+				metric: {
+					metricName: 'totalUsers',
+				},
+				desc: true,
+			},
+		],
+		limit: 3,
+	};
+
+	const topCitiesReport = useInViewSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getReport( topcCitiesReportOptions )
+	);
+
+	const loading = useInViewSelect(
+		( select ) =>
+			! select( MODULES_ANALYTICS_4 ).hasFinishedResolution(
+				'getReport',
+				[ topcCitiesReportOptions ]
+			)
+	);
+
+	const { rows = [] } = topCitiesReport || {};
+
+	const totalUsers =
+		topCitiesReport?.totals?.[ 0 ]?.metricValues?.[ 0 ]?.value;
+
+	const columns = [
+		{
+			field: 'dimensionValues',
+			Component: ( { fieldValue } ) => {
+				const [ title ] = fieldValue;
+
+				return (
+					<p className="googlesitekit-km-widget-tile__table-plain-text">
+						{ title.value }
+					</p>
+				);
+			},
+		},
+		{
+			field: 'metricValues.0.value',
+			Component: ( { fieldValue } ) => (
+				<strong>{ numFmt( fieldValue / totalUsers, '%' ) }</strong>
+			),
+		},
+	];
+
 	return (
-		<Widget>
-			<div>TODO: UI for TopCitiesWidget</div>
-		</Widget>
+		<MetricTileTable
+			Widget={ Widget }
+			title={ __( 'Top cities driving traffic', 'google-site-kit' ) }
+			loading={ loading }
+			rows={ rows }
+			columns={ columns }
+			zeroState={ ZeroDataMessage }
```

Comment:
Not unique to this issue since I see it is following others, but `zeroState` should be `ZeroState` and expect an `elementType` as is our convention. i.e. PascalCase = element type; camelCase = node.

We can fix this in a separate issue if you'd prefer, I'm just calling what I see :)

Reply:
Good catch. I have incorporated the changes in this PR to save a few hours 😃